README.md

```diff
@@ -1,17 +1,58 @@
 # Kontrol
 
+Fast Installation
+-----------------
 
-## Installation
+-   `bash <(curl https://kframework.org/install)`: install [kup package manager].
+-   `kup install kontrol`: install Kontrol.
+-   `kup list kontrol`: list available Kontrol versions.
 
-Prerequsites: `python >= 3.10`, `pip >= 20.0.2`, `poetry >= 1.3.2`.
+**NOTE**: The first run will take longer to fetch all the libraries and compile sources. (30m to 1h)
 
-```bash
-make build
-pip install dist/*.whl
+### Build from source
+
+#### K Framework
+
+You need to install the [K Framework] on your system, see the instructions there.
+The fastest way is via the [kup package manager], with which you can do to get the correct version of K:
+
+```sh
+kup install k.openssl.procps --version v$(cat deps/k_release)
+```
+
+#### Poetry dependencies
+
+First you need to set up all the dependencies of the virtual environment using Poetry with the prerequisites `python 3.8.*`, `pip >= 20.0.2`, `poetry >= 1.3.2`:
+```sh
+poetry install
+```
+
+#### Build using the virtual environment
+
+In order to build `kontrol`, you need to build these specific targets:
+```sh
+poetry run kevm-dist --verbose build -j`nproc` plugin haskell foundryx
 ```
 
+To change the default compiler:
+```sh
+CXX=clang++-14 poetry run kevm-dist --verbose build -j`nproc` plugin haskell foundryx
+```
+
+On Apple Silicon:
+```sh
+APPLE_SILICON=true poetry run kevm-dist --verbose build -j`nproc` plugin haskell foundryx
```

Comment:
`nproc` is not a command on Mac; apparently the equivalent is `sysctl -n hw.logicalcpu`, but maybe we can also recommend some fixed number of processes, e.g., `-j3`.